### ISSUE TITLE
Should bundled games be moved to an optional content snap?

### ISSUE THREAD

**Issue by mcphail** [snapcrafters/scummvm#21]
The scummvm snap has become quite large due to the bundled demos and freeware games. As previous revisions of the snap are kept around by default, installing the scummvm snap commits the user to around 1.5GB of disk space, whether or not they want to use those games. There is no way for the user to opt out of having the games installed, meaning a commitment of at least 450MB of disk space even if old revisions are purged.

Should the demos and freeware be split out? Can this be done in a content snap?

**Comment by lotharsm** [comment]
Looking at https://snapcraft.io/docs/content-interface, I think it should be possible in theory to add a new scummvm-data oder scummvm-bonus snap.

However, I'm not a huge fan of this idea - now, it's way easier for us to just update our games bundle and let the launchpad builders do the rest. I don't think 500MB of disk space are a thing these days.

EDIT: This doesn't mean I'm not looking into possible solutions for this issue though :)

**Comment by mcphail** [comment]
Yes, but as 2 previous revisions are kept by default, we're looking at 1.5GB here...

**Comment by lotharsm** [comment]
The thing is: The current way we handle this is very convenient for the average user since he get's something to play with right at the beginning - without having to install another snap.

**Comment by lotharsm** [comment]
What do you think about putting the games-bundle files inside the `$SNAP_USER_COMMON` directory? At least this would solve the duplication issue since this directory is - well - common across all revisions of the snap...

**Comment by mcphail** [comment]
You'd need to find a way to deliver them to that location, though. So you could either add them to the snap and then copy them across (but this would compound the problem as you'd have copies in the /var/lib/snapd directory and every user's home directories) or download them post-hoc on first launch. The latter option would probably need some form of gui/dialog to keep the user informed.

I've got a few snaps which download the game files post-hoc (quake-shareware, lemmings-christmas, alien-rampage and epic-pinball). Those game files are very small so the install doesn't take long. IIRC, trackmania-forever downloads post-hoc and is much bigger (and more comparable to your game bundle), so that would be an option. Would be good to have a gui option to accept or decline if you planned to take that route.

**Comment by lotharsm** [comment]
> You'd need to find a way to deliver them to that location, though. So you could either add them to the snap and then copy them across (but this would compound the problem as you'd have copies in the /var/lib/snapd directory and every user's home directories) or download them post-hoc on first launch. The latter option would probably need some form of gui/dialog to keep the user informed.
> 
> I've got a few snaps which download the game files post-hoc (quake-shareware, lemmings-christmas, alien-rampage and epic-pinball). Those game files are very small so the install doesn't take long. IIRC, trackmania-forever downloads post-hoc and is much bigger (and more comparable to your game bundle), so that would be an option. Would be good to have a gui option to accept or decline if you planned to take that route.

I think what I can do is to try to map the files into the $SNAP_USER_COMMON directory. Regarding the GUI: We (as in the ScummVM team) have some plans to bundle games and demos on our own, so this will not be a "snap-exculsive" feature in the long run. Since we are currently in the planning stage of this feature, I don't think it makes sense to modify the GUI at this point in time.
